File: python/bob/bob.py
import re


def hey(statement):
    """
    Bob the teenager limited responses to questions
    
    :param statement: input sentence or phrase string
    :return: a sentence or phrase in response
    """
    statement = re.sub(r'[^A-Za-z0-9-@!_():;.? ]', '', statement.strip())
    uppercase = statement.isupper()
    if not uppercase and statement.endswith('?'):
        return 'Sure.'
    elif statement == '':
        return 'Fine. Be that way!'
    elif uppercase:
        return 'Whoa, chill out!'
    else:
        return 'Whatever.'

File: typescript/bob/bob.ts
export default class Bob {
    hey(s: string) {
        s = s.trim()
        if (this.isUpper(s)) {
            return "Whoa, chill out!"
        } else if (s.endsWith('?')) {
            return "Sure."
        } else if (!s) {
            return "Fine. Be that way!"
        } else {
            return "Whatever."
        }
    }

    private isUpper(s: string): boolean {
        if (s === s.toUpperCase()) {
            return /[a-z]/i.test(s)
        } else {
            return false
        }
    }
}